internal/command/jsonplan/resource.go

```diff
@@ -33,6 +33,10 @@ type resource struct {
 	// unknown values are omitted or set to null, making them indistinguishable
 	// from absent values.
 	AttributeValues attributeValues `json:"values,omitempty"`
+
+	// SensitiveValues is similar to AttributeValues, but with all sensitive
+	// values replaced with true, and all non-sensitive leaf values omitted.
+	SensitiveValues map[string]bool `json:"sensitive_values,omitempty"`
```

Comment:
By using a `map[string]bool` here I think we reduce the granularity of sensitivity when compared to the precise level used in change representation (which uses a `json.RawMessage` for `BeforeSensitive`/`AfterSensitive`). This means that for the following config:

```terraform
resource "null_resource" "none" {
  triggers = {
    boop = "beep"
    flap = sensitive("honk")
  }
}
```

the `sensitive_values` output would be `{"triggers":true}` rather than `{"triggers":{"flap":true}}`. As a result, a state renderer would have to hide the entire `triggers` attribute, rather than just the one sensitive member.

This is compatible with the JSON plan sensitivity output, just more coarse. Is that necessary or could we consider keeping the full sensitivity information?

Reply:
🤔 you're absolutely correct, that ain't right, I shall fix that.